src/common/utils/ajax/helpers.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2015-2016 Imply Data, Inc.
+ * Copyright 2017-2018 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Compatibility with Q.defer or Promise.defer, now obsolete
+ */
+export class Deferred<T> {
+  resolve: (r: T) => void;
+  reject: () => void;
+  promise: Promise<T>;
+
+  constructor() {
+    this.promise = new Promise((resolve, reject) => {
+      this.resolve = resolve;
+      this.reject = reject;
+    });
+  }
+}
+
+/**
+ * used everywhere that Q(null) was used
+ */
+export function noop(): Promise<any> {
```

Comment:
I would name it something like "resolveNull" or "resolveWithNull". It's not an "no operation", it's quite meaningful operation wich could cause side effects.